Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/db/DBRequestClient.java

```diff
@@ -29,24 +28,38 @@ public void init() {
     }
     
     @Override
-    public void store(ReviewRequest reviewRequest) {
-       
+    public void store(ReviewRequest reviewRequest) {  
         client.store(reviewRequest);
         client.commit();
     }
 
-    @Override
-    public ReviewRequest getReviewRequestForConcept(String conceptId){
-        
-        ReviewRequest request = new ReviewRequest();
-        request.setConceptId(conceptId);
-        
-        List<ReviewRequest> reviewRequests = client.queryByExample(request);
+    
+    public ReviewRequest getReview(ReviewRequest request) {
+        List<ReviewRequest> responses =  client.query(new Predicate<ReviewRequest>() {
+            private static final long serialVersionUID = 6495914730735826451L;
+
+            @Override
+            public boolean match(ReviewRequest review) {
+                return review.getId().equals(request.getId());
+            }
+            
+        });
         
-        if(reviewRequests!= null && reviewRequests.size()>0) {
-            return reviewRequests.get(0);
+        if(responses == null || responses.size() == 0) {
+            return null;
         }
-        return null;
+        
+        return responses.get(0);
+    }
+    
+    @Override
+    public List<ReviewRequest> getAllReviews(ReviewRequest reviewRequest){
```

Comment:
this method should take a concept id and then build the example object here, otherwise you'll have to change the logic of the manager class again when the backend is exchanged.

Reply:
?

Reply:
?